docs/user/DockerUsage.md

```diff
@@ -2,34 +2,56 @@
 
 ## Prereqs
 
-- Make sure you have Docker [installed on your workstation](../../README.md#docker).
+- Docker [installed on your workstation](../../README.md#docker).
 
-- Prepare a file with authentication info, as described in [Authenticating Terraform to access Azure](./TerraformAzureAuthentication.md)
+## Build the Docker image
 
-## Build the docker image
-
-Run the following command to create the `viya4-iac-azure` docker image that will be used in subsequent steps:
+Run the following command to create the `viya4-iac-azure` Docker image that will be used in subsequent steps:
 
 ```bash
 docker build -t viya4-iac-azure .
 ```
 
 NOTE: The Dockerfile for the container can be found [here](../../Dockerfile).
 
-## Preparation
+## Prepare Docker Environment File for Azure Authentication
+
+Follow either one of the authentication methods described in [Authenticating Terraform to access Azure](./TerraformAzureAuthentication.md) and create a file with the authentication variable values to use with container invocation.
+
+Store these values outside of this repo in a secure file, for example
+`$HOME/.azure_docker_creds.env`. Protect that file with Azure credentials so only you have read access to it. **NOTE:** Do not use quotes around the values in the file, and make sure to avoid any trailing blanks!
+
+Now each time you invoke the container, specify the file with the [Docker `--env-file` option](https://docs.docker.com/engine/reference/commandline/run/#set-environment-variables--e---env---env-file) to pass on Azure credentials to the container e.g.
+
+```bash
+docker run <...> \
+  --env-file $HOME/.azure_docker_creds.env \
+  <...>
+```
+
+## Configure Docker Volume Mounts
 
 Add volume mounts to the `docker run` command for all files and directories that must be accessible from inside the container.
 
-The most common file reference is the value of the [`ssh_public_key`](./CONFIG-VARS.md#required-variables) variable in the `terraform.tfvars` file.
+The most common file reference is the value of the [`ssh_public_key`](../CONFIG-VARS.md#required-variables) variable in the `terraform.tfvars` file.
 
 Note that local references to `$HOME` (or "`~`") need to map to the root directory `/` in the container.
 
+You also need to mount the local directory containing your `terraform.tfvars`, that can be your current directory e.g., $(pwd). This is the same local directory where Terraform state file `terraform.tfstate` will be written. To grant Docker permission to write to the directory use [Docker `--group-add root` option](https://docs.docker.com/engine/reference/run/#additional-groups) e.g.
+
```

Comment:
I was acctually wondering about that. Isn' t it the --user option that allows the container to read/write to the local directory?
Not sure what "--group-add root" is there for, but not for write access to $pwd

Reply:
Yes I should remove that, it's to override `--user` option and to run as root